extra_foam/gui/windows/pump_probe_w.py

```diff
@@ -138,8 +148,61 @@ def initUI(self):
 
     def initConnections(self):
         """Override."""
-        pass
+        self._ctrl_widget.save_btn.clicked.connect(self.saveToFile)
 
     def closeEvent(self, QCloseEvent):
         self._ctrl_widget.resetAnalysisType()
         super().closeEvent(QCloseEvent)
+
+    def saveToFile(self):
+        # Get the current data
+        if not len(self._queue):
+            return
+        data = self._queue[0]
+
+        # Open file dialog
+        filepath, filter = QFileDialog.getSaveFileName(
+            caption="Save image",
+            directory=osp.expanduser("~"),
+            filter=f"{SaveFile.NPZ.value};;{SaveFile.TXT.value}")
+        filter = SaveFile(filter)
+
+        # Validate filepath
+        if not filepath:
+            return
+        suffix = f".{filter.name.lower()}"
+        if not filepath.lower().endswith(suffix):
+            filepath += suffix
+
+        # Copy reference file from tmp folder to desired destination
+        os.makedirs(osp.dirname(filepath), exist_ok=True)
+
+        # Save the data
+        pp = data.pp
+        train_id = data.tid
+        position = pp.x
+        intensity_on = pp.y_on
+        intensity_off = pp.y_off
+        intensity_sub = pp.y
+
+        if position is None:
+            return
```

Comment:
I think this should show an error dialog to make it clear to the user that nothing happened.